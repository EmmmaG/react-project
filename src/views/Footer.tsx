import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

const styles = (theme: Theme) => ({
	footerContainer: {
		display: 'flex',
		justifyContent: 'space-between',
		marginLeft: '35px',
		marginRight: '35px',
		paddingBottom: '20px',
	},
	copyRight: {
		color: theme.palette.grey[500],
	},
	/* eslint-disable quote-props */
	backToTop: {
		color: theme.palette.grey[600],
		'&:hover': {
			color: 'black',
			cursor: 'pointer',
		},
	},
	/* eslint-enable quote-props */
})

export interface Props extends WithStyles<typeof styles> {}

export class Footer extends React.Component<Props> {
	public render() {
		const { classes } = this.props

		return (
			<div className={classes.footerContainer}>
				<Typography className={classes.copyRight}>Copyright Â© 2018 Space Savvy</Typography>
				<Typography className={classes.backToTop}><u>Back to top</u></Typography>
			</div>
		)
	}
}

export default withStyles(styles)(Footer)
