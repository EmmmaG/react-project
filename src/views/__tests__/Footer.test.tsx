import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'

import Footer, { Props } from '../Footer'

const props: Props = {
	classes: idObj,
}

describe('Footer', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<Footer {...props} />)
		expect(wrapper).toMatchSnapshot()
	})
})
