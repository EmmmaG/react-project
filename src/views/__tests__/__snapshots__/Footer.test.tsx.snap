// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<Footer
  classes={
    Object {
      "backToTop": "Footer-backToTop-3",
      "copyRight": "Footer-copyRight-2",
      "footerContainer": "Footer-footerContainer-1",
    }
  }
/>
`;
